### ISSUE TITLE
SNOW-1739867: `snow --config-file...` does not have any effect

### ISSUE THREAD

**Issue by sfc-gh-oazarmanesh** [snowflakedb/snowflake-cli#1729]
### SnowCLI version

3.0.1

### Python version

3.11.5

### Platform

macOS-14.7-arm64-arm-64bit

### What happened

I recently upgraded snowflake-cli and tried using it again. Previous version was version 1. My config file was here: `~/.snow/config.toml`. 

- If I try testing a connection, I get a message: `FileExistsError: [Errno 17] File exists: '/Users/oazarmanesh/.snow/config.toml'`. 
- If I try `snow --config-file="/Users/oazarmanesh/.snow/config.toml"` I get this error:
```
An unexpected exception occurred. Use --debug option to see the traceback. Exception message:
[Errno 20] Not a directory: '/Users/oazarmanesh/.snow/config.toml/logs'
```
- trying this (which I know isn't valid), `snow --config-file "/Users/oazarmanesh/.snow"`  results in this error:
```
╭─ Error 
│ Invalid value for '--config-file': File '/Users/oazarmanesh/.snow' is a directory.
```
I renamed the folder to `.snowflake` but it didn't have any effect.

In short, snowflake-cli seems to always be looking in `~/Library/...` folder for the config and connections files. `snow --config-file...` does not have any effect.

### How to reproduce

My guess is this issue happens due to update from version 1 to 3.

**Comment by sfc-gh-pjob** [comment]
@sfc-gh-oazarmanesh We discussed the case via slack and it turned out that the reason of your issues was incorrect value of `SNOWFLAKE_HOME` env variable which you shouldn't need at all. Can you confirm that issue has gone and we can close the ticket?
